apps/designer/app/shared/form-utils/use-ids.ts

```diff
@@ -0,0 +1,26 @@
+import { useId } from "@webstudio-is/design-system";
+import { useRef } from "react";
+
+export const useIds = <FieldName extends string>(
+  fieldNames: readonly FieldName[]
+) => {
+  if (process.env.NODE_ENV !== "production") {
+    // It's OK to disable because process.env.NODE_ENV is not going to change
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const fieldNamesLength = useRef<number>(fieldNames.length);
+    if (fieldNamesLength.current !== fieldNames.length) {
+      throw new Error("changing fieldNames length is not supported");
```

Comment:
also in case of thrown error, rules of hooks will be broken

Reply:
Sure, but then the entire app will crash so it doesn't matter